client/src/components/Forms/CreateFieldForm/CreateFieldForm.tsx

```diff
@@ -1,55 +1,86 @@
+import { FC } from 'react';
+import { Field, Form } from 'react-final-form';
+import { useTranslation } from 'react-i18next';
+import { FormApi } from 'final-form';
 
-import FormField from '@components/FormField/FormField';
+import { errorMessageSelector } from '@redux/selectors/error';
+
+import { useAppSelector } from '@utils/hooks/useAppSelector';
 import { initialfileds } from '@utils/constants/Modules/typeSelectData';
-import { FC, useEffect, useState } from 'react';
-import { Form } from 'react-final-form';
-import { FieldCustom, FormContainer } from '../SingleFilterForm/styled';
+import { toPreviousPage } from '@utils/functions/historyBack';
 
-interface IProps {
-    selectValue:string;
-}
+import { ErrorMessage } from '@modules/Auth/styled';
+import { ButtonContainer, FormContainer, Label } from '@modules/Modules/styled';
 
-const allField = {
-    checkbox: FormField,
-    textbox: FormField,
-    textarea: FormField,
-    upload: FormField,
-    module: FormField,
-    dropdown: FormField,
-    wysiwyg: FormField,
-    map: FormField
-};
+import { Buttons } from '@components/Button/Button';
+import { FieldSettings } from '@components/Forms/Modules/FieldSettings';
+import FormField from '@components/FormField/FormField';
 
-export const CreateFieldForm: FC<IProps> = ({ selectValue }) =>{
-    // big data, need typed later 
-    const [currentField, setCurrentField] = useState<any>(); 
-    console.log(currentField);
+import { ICreateFieldProps, IFieldProps } from '@interfaces/types';
 
+import { Select } from '@modules/Settings/styled/styled';
+import { useStyles } from '@utils/styles/field';
 
-  useEffect(() => {
-    initialfileds.forEach((field) => {
-        if (field.type === selectValue){
-         setCurrentField(field);
-        } 
-     });
-  }, [selectValue]);    
+interface IProps {
+    currentField?: IFieldProps;
+    onSubmit: (value: ICreateFieldProps) => void;
+    setCurrentField: (value: IFieldProps) => void;
+}
 
-const onSubmit = (value:any) => {
-    console.log(value);
-};
-    
-return (
-    <Form
-        onSubmit={onSubmit}
-        render={({ handleSubmit }) => (
+export const CreateFieldForm: FC<IProps> = ({ currentField, onSubmit, setCurrentField }) => {
+    const { t } = useTranslation();
+    const errorMessage = useAppSelector(errorMessageSelector);
+    const classes = useStyles();
+
+    const handleChange =
+        (form: FormApi<ICreateFieldProps, Partial<ICreateFieldProps>>) =>
+        (e: React.ChangeEvent<HTMLSelectElement>) => {
+            initialfileds.forEach((field) => {
+                if (field.type === e.target.value) {
+                    setCurrentField(field);
+                }
+            });
```

Comment:
ca you use find here?

Reply:
need foreach, cos i need get field and set current type for create field form

Reply:
You can find field using arr.find(), and set it to the state. 

Currently if your array will have 1000 fields, your code will go through all of them, for what? Also if condition will be **true** more than one time? you'll change state multiple times. Also it's bad to change state inside cycle